terraform/Azure/prohibited_vm_sizes/prohibited_vm_sizes.rego

```diff
@@ -0,0 +1,17 @@
+package torque
+import input as tfplan
+
+# --- Validate azure vm sizes ---
+
+deny[reason] {
+    allowed_set:= { x | x:= data.prohibited_vm_sizes[_] }
+    results_set:= { r | r:= tfplan.resource_changes[_].change.after.vm_size }
+    diff:= results_set - allowed_set
+    
+    # print("allowed_set:       ", allowed_set)
+    # print("used_locations:    ", results_set)
+    # print("diff:              ", diff)
+
+    count(diff) > 0 # if true -> deny! and return this error ("reason") below
+    reason:= concat("",["Invalid VM size: '", sprintf("%s", [results_set]),"'. The prohibited VM sizes are: ", sprintf("%s", [allowed_set])])
```

Comment:
"Invalid **vm** size:...
The prohibited **vm**..

[Nice catch regarding the 'prohibited'.. I sent a message to Segev regarding this]